Initium-Odp/src/com/universeprojects/miniup/server/aspects/AspectSlotted.java

```diff
@@ -1,40 +1,184 @@
 		return count;
 	}
 	
-}
+	/**
+	 * Returns true if this aspect has any open slots.
+	 * @return
+	 */
+	public boolean hasOpenSlots() {
+		return getMaxCount() < getOpenCount();
+	}
+	
+	/**
+	 * Returns the number of open slots on this aspect.
+	 * @return
+	 */
+	public long getOpenCount() {
+		Long max = (Long)getProperty("maxCount");
+		
+		List<EmbeddedEntity> currentlySlotted = (List<EmbeddedEntity>)getProperty("slotItems"));
+		
+		if(currentlySlotted == null) return max;
+		
+		return max-currentlySlotted.size();
+	}
+	
+	static {
+		addCommand("InsertItemToSlot", AspectSlotted.CommandInsertItemToSlot.class);
+	}
+	
+	
+	/**
+	 * given a slotted item and a slottable item, insert it into the slot and apply the
+	 * appropriate modifiers to the parent item.
+	 * @author Evan
+	 *
+	 */
+	public static class CommandInsertItemToSlot extends Command {
+
+		public CommandInsertItemToSlot(ODPDBAccess db, HttpServletRequest request, HttpServletResponse response) {
+			super(db, request, response);
+		}
+		
+		//Pass the ID of the item with slots and the item we're slotting in. We will need to sanitize it.
+		@Override
+		public void run(Map<String, String> parameters) throws UserErrorMessage, UserRequestIncompleteException{
+			
+			//generate the base item from the given ID, then do basic checks.
+			Key baseKey = KeyFactory.createKey("Item", Long.parseLong(parameters.get("baseId")));
+			InitiumObject baseItem = db.getInitiumObject(baseKey);
+			
+			if(baseItem == null) throw new UserErrorMessage("Invalid Item ID.");
+			if(!CommonChecks.checkItemIsAccessible(baseItem.getEntity(), db.getCurrentCharacter()))
+				throw new UserErrorMessage("Nice try.");
+			
+			AspectSlotted slottedAspect = (AspectSlotted) baseItem.getInitiumAspect("Slotted");
+			
+			if(slottedAspect == null) throw new UserErrorMessage("This item has no slots.");
+			
+			if(!slottedAspect.hasOpenSlots()) throw new UserErrorMessage("All slots on this item are full.");
+			
+			//generate the slottable item from the given ID, then do basic checks.
+			Key slottableKey = KeyFactory.createKey("Item", Long.parseLong(parameters.get("slottableId")));
+			InitiumObject slottableItem = db.getInitiumObject(slottableKey);
+			
+			if(slottableItem == null) throw new UserErrorMessage("Invalid Item ID");
+			if(!CommonChecks.checkItemIsAccessible(slottableItem.getEntity(), db.getCurrentCharacter()))
+				throw new UserErrorMessage("Nice try.");
+			if(!CommonChecks.checkItemHasAspect(baseItem.getEntity(), "Slottable"))
+				throw new UserErrorMessage("This item can't be inserted into a slot.");
+			
+			//we've made it this far, which means the base item and the slottable item are both completely valid.
+			//time to actually insert the item.
+			
+			//Write a method that copies a given entity to a new embeddedentity?
+			EmbeddedEntity newEmbedded = null;
+			
+			List<EmbeddedEntity> currentlySlotted = (List<EmbeddedEntity>)baseItem.getProperty("Slotted:slotItems");
```

Comment:
Probably better to use the InitiumObject object here and get the aspect object and manipulate that instead.